### PR TITLE
Add MCP Client Connection closure

### PR DESCRIPTION
**Pull request by colindembovsky**
This pull request improves the robustness of the MCP client connection handling in `src/main.ts` and adds comprehensive tests for these scenarios. The main focus is to ensure the MCP client connection is always closed properly, handling errors gracefully, and to increase test coverage for edge cases.

### CODE REVIEW

File: src/inference.ts [hunk 1]
```diff
@@ -128,6 +133,26 @@ export async function mcpInference(
 
       if (!toolCalls || toolCalls.length === 0) {
         core.info('No tool calls requested, ending GitHub MCP inference loop')
+
+        // If we have a response format requirement and this is the final response,
+        // but it doesn't look like proper JSON, try one more time with explicit JSON request
+        if (request.responseFormat && modelResponse && iterationCount === 1) {
```
Comment (sgoedecke) [inline src/inference.ts:139]: If we're in the final response, I don't think `iterationCount` is going to equal 1. IMO we can drop that condition.

We should actually use `.response_format` to force adherence to the JSON schema, instead of politely requesting it. Maybe we need to set a `rerunningLastIteration` bool above the loop and check that in the condition on L109.

File: src/main.ts [hunk 2]
```diff
@@ -77,12 +77,28 @@ export async function run(): Promise<void> {
...
 
     if (enableMcp) {
-      const mcpClient = await connectToGitHubMCP(githubMcpToken)
+      mcpClient = await connectToGitHubMCP(githubMcpToken)
 
       if (mcpClient) {
-        modelResponse = await mcpInference(inferenceRequest, mcpClient)
+        try {
+          modelResponse = await mcpInference(inferenceRequest, mcpClient)
+        } finally {
```
Comment (sgoedecke) [inline src/main.ts:88]: Won't this close the connection immediately, before we start doing any inference?

I would expect the L126 `process.exit()` to be already closing the connections. That's where we'd want to be manually closing the client IMO.
